File: ged-web-client/src/app/main/skeleton/configuration/_model/authorisation.ts
import {Entity} from 'src/@externals/loga/_abstract';
import {Document} from "./document";
import {Dossier} from "./dossier";
import {Profile, User} from "../../../permission/_model";
import {Privilege} from "./privilege";
import {Raccourci} from "./raccourci";

export class Authorisation extends Entity<Authorisation> {

    public document: Document;
    public dossier: Dossier;
    public raccourci: Raccourci;
    public user: User;
    public privilege: Privilege;
    public profile: Profile;

    constructor() {
        super();
    }
}

File: ged-backend/src/main/java/com/loga/skeleton/domain/entity/Authorisation.java
package com.loga.skeleton.domain.entity;

import com.loga.as.entity.Profile;
import com.loga.as.entity.User;
import com.loga.bebase.entity.AbstractEntity;
import com.loga.skeleton.domain.enumeration.Privilege;
import lombok.*;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@EqualsAndHashCode(callSuper = false)
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "authorisation")
public class Authorisation extends AbstractEntity {

    @ManyToOne
    private Document document;

    @ManyToOne
    private Dossier dossier;

    @ManyToOne
    private Raccourci raccourci;

    private Privilege privilege;

    @ManyToOne
    private Profile profile;

    @ManyToOne
    private User user;

}
